coalib/bearlib/aspects/meta.py

```diff
@@ -76,23 +76,47 @@ def __repr__(cls):
         return '<%s %s>' % (type(cls).__name__, repr(cls.__qualname__))
 
 
+class aspectTypeError(TypeError):
+    """
+    This error is raised when an object is not an ``aspectclass`` or an
+    instance of ``aspectclass``
+    """
+
+    def __init__(self, item):
+        self.item = item
+        message = ('{} is not an aspectclass or an instance of an '
+                   'aspectclass'.format(repr(self.item)))
+        super().__init__(message)
+
+
+def isaspect(item):
+    """
+    This function checks whether or not an object is an ``aspectclass`` or an
+    instance of ``aspectclass``
+    """
+    return isinstance(item, (aspectclass, aspectbase))
+
+
+def assert_aspect(item):
+    """
+    This function raises ``aspectTypeError`` when an object is not an
+    ``aspectclass`` or an instance of ``aspectclass``
+    """
+    if not isaspect(item):
+        raise aspectTypeError(item)
+    return item
+
+
 def issubaspect(subaspect, aspect):
     """
     This function checks whether or not ``subaspect`` is a subaspect of
     ``aspect``.
     """
-    if (not isinstance(subaspect, aspectclass)
-            and not isinstance(subaspect, aspectbase)):
-        raise TypeError(
-            '{} is not an aspectclass or an instance of an '
-            'aspectclass'.format(repr(subaspect)))
-    if (not isinstance(aspect, aspectclass)
-            and not isinstance(aspect, aspectbase)):
-        raise TypeError(
-            '{} is not an aspectclass or an instance of an '
-            'aspectclass'.format(repr(aspect)))
-    aspect_qualname = (aspect.__qualname__ if isinstance(
-                    aspect, aspectclass) else type(aspect).__qualname__)
-    subaspect_qualname = (subaspect.__qualname__ if isinstance(
-                    subaspect, aspectclass) else type(subaspect).__qualname__)
-    return re.match(aspect_qualname+'(\.|$)', subaspect_qualname) is not None
+    if assert_aspect(subaspect) and assert_aspect(aspect):
```

Comment:
There is no `if` condition needed since `assert_aspect` either `raise`s or just returns its argument